iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/json/ExpressionImpl.java

```diff
@@ -26,21 +26,14 @@
 import org.sonar.iac.common.api.tree.impl.TextRange;
 import org.sonar.iac.common.yaml.tree.YamlTreeMetadata;
 
-public class ExpressionImpl extends AbstractArmTreeImpl implements Expression {
+public abstract class ExpressionImpl extends AbstractArmTreeImpl implements Expression {
 
-  private final String value;
   private final YamlTreeMetadata metadata;
 
-  public ExpressionImpl(String value, YamlTreeMetadata metadata) {
-    this.value = value;
+  protected ExpressionImpl(YamlTreeMetadata metadata) {
     this.metadata = metadata;
   }
 
-  @Override
-  public String value() {
-    return value;
-  }
-
   @Override
```

Comment:
I think we should not implement this here, since it needs to be overridden in an implementing class.